lib/src/hidden_ranges/hidden_line_ranges_builder.dart

```diff
@@ -0,0 +1,69 @@
+import '../code/code_lines.dart';
+import 'hidden_line_ranges.dart';
+import 'hidden_ranges.dart';
+import 'line_numbering_breakpoint.dart';
+
+class HiddenLineRangesBuilder {
+  final HiddenLineRanges hiddenLineRanges;
+
+  factory HiddenLineRangesBuilder({
```

Comment:
Also I'd like to convert it to a method

Reply:
We have a pattern in `HiddenRangesBuilder`. This piece follows it.

The reason it is a class and not a method is that we may add more fields as we have in `HiddenRangesBuilder`.